specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-04-01-preview/costmanagement.json

```diff
@@ -1796,6 +1796,786 @@
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/View"
+            },
+            "description": "Parameters supplied to the CreateOrUpdate View operation."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ViewResult"
+            }
+          },
+          "201": {
+            "description": "Created.",
+            "schema": {
+              "$ref": "#/definitions/ViewResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Views"
+        ],
+        "operationId": "Views_DeleteByExternalSubscription",
+        "description": "The operation to delete a view by external subscription.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/costmanagement/"
+        },
+        "x-ms-examples": {
+          "ExternalSubscriptionDeleteView": {
+            "$ref": "./examples/View/ViewDelete/ViewDeleteByExternalSubscription.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/scopeViewsParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/viewNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.CostManagement/checkNameAvailability": {
+      "post":{
+        "tags":[
+          "CostManagement"
+        ],
+        "operationId": "CostManagement_CheckNameAvailability",
+        "description": "The operation to check resource name availability under cost management. A name is available if no other resource exists with the same name, type and scope.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/costmanagement/"
+        },
+        "x-ms-examples": {
+          "SubscriptionCheckNameAvailabilityAvailable": {
+            "$ref": "./examples/CheckNameAvailability/Available/CheckNameAvailabilityBySubscription_Available.json"
+          },
+          "SubscriptionCheckNameAvailabilityNotAvailable": {
+            "$ref": "./examples/CheckNameAvailability/NotAvailable/CheckNameAvailabilityBySubscription_NotAvailable.json"
+          },
+          "ResourceGroupCheckNameAvailabilityAvailable": {
```

Comment:
Do we need so many examples that are only differ by scope? This is crazy redundant and will only drown out the value of other examples. Can we only have the subscription examples?

Reply:
Have to separately list two scopes with external billing account and external subscription, as their routing templates are not compatible with the existing one.